### PR TITLE
fix bugs to allow build with gcc 7.2.0 under Ubuntu 17.10

### PR DESCRIPTION
**Pull request by asanoaozora**
Fixes #[3]

This PR is **[ready]** for review.

### Risk
This PR makes **[no]** API changes.

### Testing Plan
[After successful built, /.start.sh  and chromium-browser index.html, the html page displayed]

### Summary
[To build it under Ubuntu with gcc 7.2.1, somes fix have been necessary to apply]

##### Breaking Changes
* [Breaking change info]

##### Enhancements
* [Enhancement info]

##### Bug Fixes
* [Error on apache_log4cxx fixed by applying patches of the version log4cxx_0.10.0-11.debian.tar.xz]
*[enum returned in a bool->replaced by false]
*[unused but set variable -> commented]

### Tasks Remaining:
- [ ] [Task 1]
- [ ] [Task 2]

### CLA
- [ ] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/src/policies/policy_handler.cc [hunk 1]
```diff
@@ -1768,7 +1768,7 @@ void PolicyHandler::OnCertificateUpdated(const std::string& certificate_data) {
 void PolicyHandler::OnPTUFinished(const bool ptu_result) {
   LOG4CXX_AUTO_TRACE(logger_);
   sync_primitives::AutoLock lock(listeners_lock_);
-  HandlersCollection::const_iterator it = listeners_.begin();
+//  HandlersCollection::const_iterator it = listeners_.begin();
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/policies/policy_handler.cc]: @asanoaozora please remove unused code. Do not leave it commented.

**Comment by asanoaozora** [comment]
Done, rebuilt, launch the core and the html page successfully, except these messages size inconsistency (?)
$ ./start.sh 
accept_buf: /Ekd//GI6983yqQUZB4Ajw8ItSg=
accept_buf: z0o8u0AunK4Fjgjp1cHlvyx3/J4=
accept_buf: gnwd3huJXT2/cCSyM1XS8jm+sXY=
accept_buf: 3NdM0dp28sRnU/cuqkF4aTXz5uM=
accept_buf: J4+v2cg6DIvndegKJxhvWVDAU/4=
accept_buf: 9pBIrHTQVOxILXDOmr1r6izAIao=
accept_buf: 9N8CdO5ZbmChCxxyM9yqGFOyOTM=
accept_buf: Sg6+eMckqBUSYiF9gmG5k1LxVvk=
ERROR!!! ~/github/sdl_core/src/3rd_party-static/message_broker/src/server/mb_tcpserver.cpp:96 Received 4096 actual data length 4916
ERROR!!! ~/github/sdl_core/src/3rd_party-static/message_broker/src/server/mb_tcpserver.cpp:97 Incomplete message
ERROR!!! ~/github/sdl_core/src/3rd_party-static/message_broker/src/server/mb_tcpserver.cpp:96 Received 4096 actual data length 4916
^CERROR!!! ~/github/sdl_core/src/3rd_party-static/message_broker/src/server/mb_tcpserver.cpp:97 Incomplete message

**Comment by jacobkeeler** [comment]
@asanoaozora Please sign the CLA and mark it as complete. Also, please reference an issue that this change fixes (I don't know where you got #3 from, but it needs to be an issue within this repository), it looks like this may resolve https://github.com/smartdevicelink/sdl_core/issues/1495

**Comment by jacobkeeler** [comment]
Closing in favor of #2097

**Pull request closed without merging**